src/workers/postUpdated.ts

```diff
@@ -472,19 +511,33 @@ const worker: Worker = {
             mergedKeywords,
             questions,
           });
+
+          postId = newPost?.id;
         } else {
           // Handle update of existing post
           await updatePost({
             counter,
             logger,
             entityManager,
             data: fixedData,
-            id: post_id,
+            id: postId,
             mergedKeywords,
             questions,
             content_type,
           });
         }
+
+        if (postId) {
+          await handleCollectionRelations({
```

Comment:
This is because it needs to be for both creation/update?
Maybe we can rather have it inside both functions?

Just thinking because it's just a bit harder to follow now.

Reply:
@rebelchris I suggest logging the need for a refactor here. I agree we can make it a bit more straightforward

Reply:
Generally I see the need to handle different post types in the future in structured way because if you look at my changes I had to add `content` and different properties to `fixData` and using union types.

Reply:
Yeah it started as one type, but as we grow it's very messy.
Agreed, we should probably split it up a bit better, and even use different type handlers.

WIll log it, no action needed.

Reply:
👌 tnx!